pypeit/core/gui/object_find.py

```diff
@@ -0,0 +1,684 @@
+        prof, _ = np.histogram(coords[ww], bins=edges, weights=self.frame[ww])
+        return prof/ww[0].size
+
+    def set_fwhm(self, xdata):
+        """Set the FWHM using the available panel
+
+        Args:
+            xdata (float): The x coordinate selected by the user
+        """
+        self.specobjs[self._obj_idx].fwhm = 2.0*np.abs(xdata)
+        self.draw_profile()
+        self.replot()
+        return
+
+    def get_specobjs(self):
+        """Get the updated version of SpecObjs
+
+        Returns:
+            sobjs (SpecObjs): SpecObjs Class
+        """
+        if self._use_updates:
+            return self.specobjs
+        else:
+            return None
+
+    def update_infobox(self, message="Press '?' to list the available options",
+                       yesno=True, default=False):
+        """Send a new message to the information window at the top of the canvas
+
+        Args:
+            message (str): Message to be displayed
+        """
+        self.axes['info'].clear()
+        if default:
+            self.axes['info'].text(0.5, 0.5, "Press '?' to list the available options", transform=self.axes['info'].transAxes,
+                          horizontalalignment='center', verticalalignment='center')
+            self.canvas.draw()
+            return
+        # Display the message
+        self.axes['info'].text(0.5, 0.5, message, transform=self.axes['info'].transAxes,
+                      horizontalalignment='center', verticalalignment='center')
+        if yesno:
+            self.axes['info'].fill_between([0.8, 0.9], 0, 1, facecolor='green', alpha=0.5, transform=self.axes['info'].transAxes)
+            self.axes['info'].fill_between([0.9, 1.0], 0, 1, facecolor='red', alpha=0.5, transform=self.axes['info'].transAxes)
+            self.axes['info'].text(0.85, 0.5, "YES", transform=self.axes['info'].transAxes,
+                          horizontalalignment='center', verticalalignment='center')
+            self.axes['info'].text(0.95, 0.5, "NO", transform=self.axes['info'].transAxes,
+                          horizontalalignment='center', verticalalignment='center')
+        self.axes['info'].set_xlim((0, 1))
+        self.axes['info'].set_ylim((0, 1))
+        self.canvas.draw()
+
+    def empty_mantrace(self):
+        """Generate an empty dictionary for the manual tracing
+        """
+        self._mantrace = dict(spat_a=[], spec_a=[], spat_trc=None, spec_trc=np.arange(self.nspec), polyorder=0)
+        return
+
+
+def initialise(frame, trace_dict, sobjs, slit_ids=None):
+    """Initialise the 'ObjFindGUI' window for interactive object tracing
+
+        Args:
+            frame (ndarray): Sky subtracted science image
+            trace_dict (dict): Dictionary containing slit and object trace information
+            sobjs (SpecObjs): SpecObjs Class
+            slit_ids (list, None): List of slit ID numbers
+
+        Returns:
+            ofgui (ObjFindGUI): Returns an instance of the ObjFindGUI class
```

Comment:
nit:  Kyle prefers we do:

```
ObjFindGui:  desc
```

i.e. no naming the variable

Reply:
OK - thanks for pointing that out - I've fixed all instances of this in the PR!